### ISSUE TITLE
pact-jvm-server (3.5.15++): Internal Server Error on /complete

### ISSUE THREAD

**Issue by hstamminger** [DiUS/pact-jvm#845]
Hi

We try to use the Pact-JVM-Server Version 3.5.15++ (up to current 3.6.1). We are using JDK 8 (1.8.0_201-b09) for running the server.

When we call "/complete" on the Pact-JVM-Server we always get an "Internal Server Error". In the Pact-JVM-Server console-log we see the following exception: 
```
Exception caught handling request:
java.lang.ClassCastException: java.lang.Integer cannot be cast to java.lang.String
        at au.com.dius.pact.server.Complete$.getPort(Complete.scala:13)
        at au.com.dius.pact.server.Complete$.apply(Complete.scala:28)
        at au.com.dius.pact.server.RequestRouter$.dispatch(RequestRouter.scala:36)
        at au.com.dius.pact.server.RequestHandler.handle(RequestHandler.scala:24)
        at au.com.dius.pact.server.RequestHandler.$anonfun$intent$1(RequestHandler.scala:28)
        at scala.PartialFunction$$anonfun$apply$1.applyOrElse(PartialFunction.scala:249)
        at scala.PartialFunction$Lifted.apply(PartialFunction.scala:224)
        at scala.PartialFunction$Lifted.apply(PartialFunction.scala:220)
        at unfiltered.response.Pass$PartialAttempt.attemptWithPass(pass.scala:72)
        at unfiltered.response.Pass$PassingAttempt.attempt(pass.scala:65)
        at unfiltered.response.Pass$PassingAttempt.attempt$(pass.scala:65)
        at unfiltered.response.Pass$PartialAttempt.attempt(pass.scala:67)
        at unfiltered.response.Pass$.$anonfun$fold$1(pass.scala:48)
        at unfiltered.response.Pass$FunctionAttempt.apply(pass.scala:77)
        at unfiltered.netty.cycle.Plan.$anonfun$channelRead$3(plans.scala:68)
        at unfiltered.netty.cycle.Plan.catching(plans.scala:41)
        at unfiltered.netty.cycle.Plan.catching$(plans.scala:40)
        at au.com.dius.pact.server.RequestHandler.catching(RequestHandler.scala:17)
        at unfiltered.netty.cycle.Plan.$anonfun$channelRead$2(plans.scala:67)
        at unfiltered.netty.cycle.SynchronousExecution.executeIntent(deferrals.scala:23)
        at unfiltered.netty.cycle.SynchronousExecution.executeIntent$(deferrals.scala:23)
        at au.com.dius.pact.server.RequestHandler.executeIntent(RequestHandler.scala:17)
        at unfiltered.netty.cycle.Plan.$anonfun$channelRead$1(plans.scala:66)
        at unfiltered.netty.cycle.Plan.catching(plans.scala:41)
        at unfiltered.netty.cycle.Plan.catching$(plans.scala:40)
        at au.com.dius.pact.server.RequestHandler.catching(RequestHandler.scala:17)
        at unfiltered.netty.cycle.Plan.channelRead(plans.scala:65)
        at unfiltered.netty.cycle.Plan.channelRead$(plans.scala:61)
        at au.com.dius.pact.server.RequestHandler.channelRead(RequestHandler.scala:17)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340)
        at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:102)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340)
        at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:323)
        at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:297)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340)
        at io.netty.channel.ChannelInboundHandlerAdapter.channelRead(ChannelInboundHandlerAdapter.java:86)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340)
        at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1434)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
        at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:965)
        at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:163)
        at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:644)
        at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:579)
        at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:496)
        at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:458)
        at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:897)
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
        at java.lang.Thread.run(Thread.java:748)
``` 

Here are the requests we made:
**create mock**
*request*
```
curl --request POST \
  --url 'http://localhost:29999/create?state=test&path=' \
  --header 'content-type: application/json' \
  --data '{
  "consumer": {
    "name": "pact-consumer"
  },
  "provider": {
    "name": "pact-provider"
  },
  "interactions": [
    {
      "description": "get time",
      "request": {
        "method": "GET",
        "path": "/resource/time/current",
	"query": {
	  "locale": ["de"]
	}
      },
      "response": {
        "body": {
          "time": "13:00:34.890"
         },
        "status": 200
      },
      "providerStates": [
        {
          "name": "test"
        }
      ]
    }
  ],
  "metadata": {
    "pact-specification": {
      "version": "3.0.0"
    }
  }
}
'
``` 
*response*
```
{"port": 32695}
``` 

**complete pact mock**
*request* 
``` 
curl --request POST \
  --url http://localhost:29999/complete \
  --header 'content-type: application/json' \
  --data '{
  "port": 32695
}'
```
*response* 
``` Internal Server Error ```

**Comment by uglyog** [comment]
Fixed with commit 136c69bc

**Comment by uglyog** [comment]
This has been released with version 3.6.2
